### ISSUE TITLE
Cannot link Swift package dependency TOCropViewController into tuist project

### ISSUE THREAD

**Issue by tackgyu** [tuist/tuist#4606]
## Describe the bug and the expected behavior

I'm trying to link the following library to our project:
https://github.com/TimOliver/TOCropViewController

It offers two products via Package.swift: one is TOCropViewController which is written in Objective-C and the other is CropViewController which consists of Swift interfaces bridging its companion.

- By linking TOCropViewController the project compiles, but Objective-C interfaces cannot be referenced in Swift code.
- By linking CropViewController the project does not compile.

Tested in tuist version 3.7.0.

## Reproduction steps

Steps to reproduce the bug:

1. `tuist init Test --platform ios`.
2. `cd Test`.
3. `touch Tuist/Dependencies.swift`
4. Add the following content to `Dependencies.swift`:
```swift
import ProjectDescription

let depenedencies = Dependencies(
    swiftPackageManager: .init([
        .remote(url: "https://github.com/TimOliver/TOCropViewController", requirement: .exact("2.6.1"))
    ]),
    platforms: [.iOS]
)
```
5. Link the lib to app as external dependency:
```swift
let mainTarget = Target(
    ...
    dependencies: dependencies + [.external(name: "TOCropViewController")]
)
```
6. `tuist generate`
7. Build project

### Error log

Bunch of error logs complaining it cannot find Objective-C symbols.

### Additional context

- macOS version: 12.4
- Tuist version: 3.7.0
- Xcode version: 13.4.1

**Comment by danyf90** [comment]
Can't replicate, please provide a sample project

**Comment by sbo-nemlig** [comment]
We are having the same issues here is a sample project  https://github.com/sbo-nemlig/testCropViewController

**Comment by tackgyu** [comment]
Thanks @sbo-nemlig and sorry for late response @danyf90.

Here's another sample project: https://github.com/tackgyu/tocropviewcontroller-compile-error

**Comment by Schlabbi** [comment]
Any update on this? Having the same problem.

**Comment by nickaroot** [comment]
You can fix it on Project level

**Fix (Tuist 4.6.0):**

Add _HEADER_SEARCH_PATHS_, _OTHER_CFLAGS_ and _OTHER_SWIFT_FLAGS_  to any target importing TOCropViewController like this:


```swift
let project = Project(
    ...
    settings: .settings(
        base: [
            "HEADER_SEARCH_PATHS": [
                "$(SRCROOT)/Tuist/.build/checkouts/TOCropViewController/Objective-C/TOCropViewController/include/TOCropViewController",
            ],
            "OTHER_CFLAGS": [
                "-fmodule-map-file=$(SRCROOT)/Tuist/.build/tuist-derived/TOCropViewController/TOCropViewController.modulemap",
            ],
            "OTHER_SWIFT_FLAGS": [
                "-Xcc",
                "-fmodule-map-file=$(SRCROOT)/Tuist/.build/tuist-derived/TOCropViewController/TOCropViewController.modulemap",
            ],
        ],
        ...
    )
    ...
)
```
